src/templating/content.md

```diff
@@ -0,0 +1,654 @@
+        value: dev
+```
+
+If you run kustomize this time, the output will be different:
+
+```terminal|command=1|title=bash
+kubectl kustomize dev
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test-pod
+spec:
+  containers:
+  - image: envoyproxy/envoy:v1.12.2
+    name: proxy-container
+    ports:
+    - containerPort: 80
+  - env:
+    - name: ENV
+      value: dev
+    image: k8s.gcr.io/busybox
+    name: test-container
+```
+
+Please notice how:
+
+1. you still have the extra container which is patched in the base `kustomization.yaml` file
+1. the environment variable was changed to `dev`
+
+You can imagine that you can have more folders and more `kustomization.yaml` to match your clusters and environments.
+
+_Can you also edit fields without having to create a folder, a `kustomization.yaml` and the extra YAML?_
+
+Kustomize supports inline editing for some fields.
+
+Kustomize is more structured that `yq` but this convenience comes with some drawbacks.
+
+Kustomize is designed on purpose to [make it hard to change values through the command line](https://github.com/kubernetes-sigs/kustomize/blob/master/docs/eschewedFeatures.md#unstructured-edits).
+
+You have to write YAML files to change even a single value — unlike `yq`.
+
+_Is there an alternative to Kustomize and `yq` that is flexible and structured?_
+
+## Writing YAML with code
+
+You could generate YAML programatically with code.
+
+And you don't even need to start from scratch.
+
+Kubernetes has several official libraries where you can create objects such as Deployments and Pod with code.
+
+> Please note that since Kubernetes uses an OpenAPI, you can autogenerate a library for your favourite language — if there isn't one already.
+
+As an example, let's have a look at how you can use Javascript to create a Pod and programatically change it.
+
+```js|title=pod.js
+const { Pod, Container } = require('kubernetes-models/v1')
+
+const pod = new Pod({
+  metadata: {
+    name: 'test-pod',
+  },
+  spec: {
+    containers: [
+      new Container({
+        name: 'test-container',
+        image: 'k8s.gcr.io/busybox',
+        env: [{ name: 'ENV', value: 'production' }],
+      }),
+    ],
+  },
+})
+
+console.log(JSON.stringify(pod, null, 2))
+```
+
+You can execute the script with the `node` binary:
+
+```terminal|command=1|title=bash
+node pod.js
+{
+  "metadata": {
+    "name": "test-pod"
+  },
+  "spec": {
+    "containers": [
+      {
+        "name": "test-container",
+        "image": "k8s.gcr.io/busybox",
+        "env": [
+          {
+            "name": "ENV",
+            "value": "production"
+          }
+        ]
+      }
+    ]
+  },
+  "apiVersion": "v1",
+  "kind": "Pod"
+}
+```
+
+The output is a Pod, but it's in JSON and not YAML.
+
+That shouldn't be a problem because:
+
+1. YAML is a superset of JSON. Any JSON file is also a valid YAML file
+1. The Kubernetes API receives the resource in JSON even if you write YAML files. `kubectl` serialises the YAML into JSON
```

Comment:
Change this to something like:

>The Kubernetes API server accepts resource definitions in both JSON and YAML. YAML is often used when resource definitions are created manually, whereas JSON is often used when resource definitions are generated automatically.

Reply:
Is this true? I thought the serialisation to JSON happens in kubectl. I cannot do a curl request with some YAML to the API?